util-hadoop/src/main/java/com/google/cloud/hadoop/util/CredentialFromAccessTokenProviderClassFactory.java

```diff
@@ -115,6 +118,34 @@ public static Credential credential(
     return null;
   }
 
+  /**
+   * Generate a {@link Credential} from the internal access token provider based on the service
+   * account to impersonate.
+   */
+  public static Credential credentialFromImpersonationServiceAccount(
```

Comment:
Change return type to `Optional<Credential>`

Reply:
Done.